src/SixLabors.Fonts/ArrayBuilder{T}.cs

```diff
@@ -153,23 +154,111 @@ private void EnsureCapacity(int min)
                 newCapacity = (uint)min;
             }
 
-            var array = new T[newCapacity];
-
+            T[] array = ArrayPool<T>.Shared.Rent((int)newCapacity);
             if (this.size > 0)
             {
                 Array.Copy(this.data!, array, this.size);
+                ArrayPool<T>.Shared.Return(this.data!);
             }
 
             this.data = array;
         }
     }
 
+    public readonly bool Contains(T item)
+    {
+        for (int i = 0; i < this.size; i++)
+        {
+            if (this[i].Equals(item))
```

Comment:
This introduces boxing which incurs allocating memory on the heap.

Reply:
I have copy the function from `List<T>.Contains` for this part. Should I finish this part?

Reply:
Using memory pooling for this type is not a good idea. 

It’s problematic because it allows unrestricted types, leading to inefficient pooling and increased risk of pool exhaustion. Each type parameter combination creates a unique pool, fragmenting the memory and reducing reuse. If frequently requested types deplete their pools, allocations bypass pooling, negating performance benefits and increasing GC pressure.

That’s why the type does not already exist in the runtime and why our memory allocators have explicit constraints.

Reply:
The reason I do that:
1. ArrayPool will return the larger array inside or create new array for that.
2. ArrayPool will delete the array object when something rent it from pool, and add it back when return it.
And also, this type(struct) is internal type. So where it used and what the unrestricted types used are also knowned.
Everything in there are manual assign.